### PR TITLE
chromedriver_update blocks "geordi rspec" from working offline

### PR DESCRIPTION
**Pull request by brunosedler**

### CODE REVIEW

File: lib/geordi/chromedriver_updater.rb [hunk 1]
```diff
@@ -94,7 +105,7 @@ def chromedriver_url(chrome_version)
         if chromedriver && chromedriver["url"]
           chromedriver["url"]
         else
-          Interaction.fail("Could not find chromedriver download url for Chrome version v#{chrome_version}.")
+          raise ProcessingError, "Could not find chromedriver download url for Chrome version v#{chrome_version}."
```
Comment (codener) [inline lib/geordi/chromedriver_updater.rb:108]: Please change all output to speak of "Chrome $version", e.g. "Chrome 119.2344.1"

File: lib/geordi/chromedriver_updater.rb [hunk 2]
```diff
@@ -81,8 +87,13 @@ def fetch_response(url, error_message)
...
         yield(response)
       else
-        Interaction.fail(error_message)
+        raise ProcessingError, error_message
       end
+
+    # Rescue Errno::NOERROR, Errno::ENOENT, Errno::EACCES, Errno::EFAULT, Errno::ECONNREFUSED, Errno::ECONNABORTED, Errno::ECONNRESET, Errno::EHOSTDOWN, Errno::EHOSTUNREACH, ...,
+    # all of which are a subclass of SystemCallError
+    rescue SystemCallError => e
+      raise ProcessingError, "Could not connect to Server: #{e.message}"
```
Comment (codener) [inline lib/geordi/chromedriver_updater.rb:96]: Nice rescue :muscle: 

I think we'd need to be a little more abstract here though:
```suggestion
      raise ProcessingError, "Request failed: #{e.message}"
```

File: lib/geordi/chromedriver_updater.rb [hunk 3]
```diff
@@ -6,9 +6,12 @@
 
 module Geordi
   class ChromedriverUpdater
+    class ProcessingError < StandardError; end
+
     VERSIONS_PER_MILESTONES_URL = "https://googlechromelabs.github.io/chrome-for-testing/latest-versions-per-milestone-with-downloads.json"
 
     def run(options)
+      interaction_method = (options[:exit_on_failure] == true) ? :note : :fail
```
Comment (codener) [inline lib/geordi/chromedriver_updater.rb:14]: I'd like to see this in the rescue block, if possible. Also, I'd change it to:
```suggestion
      interaction_method = options.exit_on_failure? ? :fail : :warn
```
/edit: the syntax `options.exit_on_failure?` might only work within Thor or with a Thor options hash. If it complicates Test, you may as well stick to your classic syntax – but please mind the twisted method names!

File: spec/chromedriver_updater_spec.rb [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+      it 'exits the command with status code 1 if an error occurs' do
+        allow(subject).to receive(:determine_chrome_version).and_raise(Geordi::ChromedriverUpdater::ProcessingError, 'Could not determine the version of Google Chrome.')
+        expect(Geordi::Interaction).to receive(:fail).with('Could not determine the version of Google Chrome.')
+
+        subject.run({})
+      end
+    end
+
+    context 'exit_on_failure=false' do
+      it 'prints an interaction note if an error occurs' do
```
Comment (codener) [inline spec/chromedriver_updater_spec.rb:13]:
```suggestion
      it 'only prints a warning if an error occurs' do
```

File: spec/chromedriver_updater_spec.rb [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+        allow(subject).to receive(:determine_chrome_version).and_raise(Geordi::ChromedriverUpdater::ProcessingError, 'Could not determine the version of Google Chrome.')
```
Comment (codener) [inline spec/chromedriver_updater_spec.rb:14]: When stubbing, I like to use custom return values to make the connection between stub and expectation clear. E.g.
```suggestion
        allow(subject).to receive(:determine_chrome_version).and_raise(Geordi::ChromedriverUpdater::ProcessingError, 'The error message')
```
Above as well.

File: spec/chromedriver_updater_spec.rb [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+        expect(Geordi::Interaction).to receive(:note).with('Could not determine the version of Google Chrome.')
+
+        subject.run(exit_on_failure:false)
```
Comment (codener) [inline spec/chromedriver_updater_spec.rb:17]: I have no clue how these tests could be green with the present code, as method names are twisted.

Comment (brunosedler) [reply spec/chromedriver_updater_spec.rb:17]: Good question. I might have done some refactoring after writing the tests and not forgot to run them again. 
And it seems our github actions test setup only runs feature tests. We need to change that.

**Review by codener — APPROVED** [review]
LGTM!

**Pull request closed without merging**